src/Tools/ExternalAccess/LiveShare/RemoteLanguageServiceWorkspaceHost.cs

```diff
@@ -111,6 +117,16 @@ public async Task EnsureProjectsLoadedAsync(CancellationToken cancellationToken)
             }
         }
 
+        /// <summary>
+        /// Initialize the workspace.  This must be done on the UI thread because it
+        /// could potentially attempt to create services that require the UI thread. <see cref="LanguageSettingsPersister"/>
```

Comment:
could we instead make those other services *not* require the UI thread?  we've been doing a lot of work to get things off the UI, it would unfortunate to slide back there.

Reply:
making LanguageSettingPersister (or IOptionPersister) UI thread free is not small work.

I believe proper way to fix it is make IOptionPersister (http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Options/IOptionPersister.cs,b13bd3bcaf4949ad) to be async interface.

and change everyone who using it to async

and then make LanguageSettingPersister to use JTF to lazily initialize it when they need it.

and caller (directly or indirectly through other service) on UI thread should change to use JTF.Run

…

any other way to fix it seems a hack such as creating yet another service just for LanguageSettingPersister so that it can queue actual work to other service which can have race (if someone ask option too fast, data is not there yet or if someone save data at wrong time, it doesn't get saved since VS passed shutdown time, or block shutdown time and etc which just make things more complicated)

…

anyway, I believe this PR is not trying to fix underlying option persister problem we have been having for several times now. 

we should have issue opened for this issue and there are other IOptionPersister issue (https://github.com/dotnet/roslyn/issues/27677, https://github.com/dotnet/roslyn/issues/27678)

and fix it all together. and this part of code (UI, JTF) seems people having many different opinions.

so, we should bring this to design meeting and agree on how to fix option persister issue once for all. 

tagging @jinujoseph

Reply:
meanwhile, I don't believe we should block this PR leaving liveshare in broken state. tagging @jinujoseph